compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -723,7 +751,12 @@ public Optional<TypeSymbol> transform(TableConstructorExpressionNode node) {
                         requireNonNullElse(((BTableType) referredType).constraint, expectedType)));
             }
 
-        } else if (isWithinParenthesis(linePosition, node.openBracket(), node.closeBracket())) {
+        }
+
+        Token openParen = node.openBracket();
+        Token closeParen = node.closeBracket();
+        if (!isParenthesisMissing(openParen, closeParen) &&
+                isWithinParenthesis(linePosition, openParen.lineRange(), node.closeBracket().lineRange())) {
```

Comment:
close-bracket is intitialized we can use it instead of `node.closeBracket()`

Reply:
As a second thought, since we don't have a popular common name for `[ ]`, `{ }`, `( )`, its totally fine going with the current naming convention